stub/src/main/java/io/grpc/stub/ServerCalls.java

```diff
@@ -332,13 +332,16 @@ public void setCompression(String compression) {
     @Override
     public void onNext(RespT response) {
       if (cancelled) {
-        throw Status.CANCELLED.withDescription("call already cancelled").asRuntimeException();
-      }
-      if (!sentHeaders) {
-        call.sendHeaders(new Metadata());
-        sentHeaders = true;
+        if (onCancelHandler == null) {
```

Comment:
nit: Could this have an `else` that `return`s? That would allow removing the current `else` and returning the code to look a bit more like it did before. (I do see that this code as written mirrors onCompleted, but it also has a single line in the else, so it doesn't matter as much.)

Reply:
Addressed in 8276d64.